decidim-meetings/app/controllers/decidim/meetings/directory/meetings_controller.rb

```diff
@@ -46,19 +35,57 @@ def default_filter_params
           {
             date: "upcoming",
             search_text: "",
-            scope_id: "",
-            space: "all"
+            activity: "all",
+            scope_id: default_filter_scope_params,
+            space: default_filter_space_params,
+            type: default_filter_type_params,
+            origin: default_filter_origin_params,
+            category_id: default_filter_category_params
           }
         end
 
+        def default_filter_category_params
+          participatory_spaces = current_organization.public_participatory_spaces
+          list_of_ps = []
+          participatory_spaces.flat_map do |current_participatory_space|
+            next unless current_participatory_space.respond_to?(:categories)
+
+            key_point = current_participatory_space.class.name.gsub("::", "__") + current_participatory_space.id.to_s
+
+            list_of_ps.push(key_point)
+            list_of_ps += current_participatory_space.categories.pluck(:id).map(&:to_s)
+          end
+
+          ["all"] + list_of_ps
+        end
+
+        def default_filter_space_params
+          %w(all) + current_organization.public_participatory_spaces.collect(&:model_name).uniq.collect(&:name).collect(&:underscore)
+        end
+
+        def default_filter_type_params
+          %w(all) + Decidim::Meetings::Meeting::TYPE_OF_MEETING
+        end
+
+        def default_filter_scope_params
+          %w(all global) + current_organization.scopes.pluck(:id).map(&:to_s)
+        end
+
         def default_search_params
           {
             scope: Meeting.not_hidden.visible_meeting_for(current_user)
           }
         end
 
+        def default_filter_origin_params
```

Comment:
💬 Do you think it makes sense to move these three methods to a concern (`default_filter_type_params`, `default_search_params` and `default_filter_origin_params`) to be more DRY reusing it here and in the "regular" meetings controller?
I was going to ignore it, but I think it could be a good idea to help future improvements to have a common place to put shared logic between the two pages.